### PR TITLE
Fix array docs not showing up in openapi emitter

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix https://github.com/microsoft/typespec/issues/2002

### CODE REVIEW

File: packages/compiler/lib/lib.tsp [hunk 1]
```diff
@@ -124,18 +124,18 @@ scalar boolean;
  * Represent a model
  */
 // Deprecated June 2023 sprint
-@deprecated("object is deprecated. Please use {} for an empty model, `Record<unknown>` for a record with unknown propertie types, `unknown[]` for an array.")
+@deprecated("object is deprecated. Please use {} for an empty model, `Record<unknown>` for a record with unknown property types, `unknown[]` for an array.")
 model object {}
 
 /**
- * Array model type, equivalent to `T[]`
+ * @dev Array model type, equivalent to `T[]`
```
Comment (markcowl) [inline packages/compiler/lib/lib.tsp:131]: What does the @dev tag signify?

File: packages/openapi3/src/types.ts [hunk 2]
```diff
@@ -431,7 +431,7 @@ export type OpenAPI3Schema = Extensions & {
    *
    * @see https://tools.ietf.org/html/draft-wright-json-schema-validation-01#section-6.23
    */
-  enum?: string[];
+  enum?: (string | number | boolean)[];
```
Comment (markcowl) [inline packages/openapi3/src/types.ts:434]: Why is this changee here?

**Review by markcowl — CHANGES_REQUESTED** [review]
One question, otherwise lgtm

File: packages/openapi3/src/types.ts (hunk 2 above)
Comment (timotheeguerin) [reply packages/openapi3/src/types.ts:434]: I added a type check in the openapi.ts and this type wasn't accurate

File: packages/compiler/lib/lib.tsp (hunk 1 above)
Comment (timotheeguerin) [reply packages/compiler/lib/lib.tsp:131]: It is a current workaround so this doc doesn't carry over  to `model foo is x[]`

I filed an issue so we can discuss this pattern https://github.com/microsoft/typespec/issues/2003

**Review by markcowl — APPROVED** [review]

**Pull request merged**